modules/post/multi/recon/local_exploit_suggester.rb

```diff
@@ -74,32 +109,53 @@ def set_module_options(mod)
   end
 
   def is_module_wanted?(mod)
-    (
-      mod.kind_of?(Msf::Exploit::Local) &&
-      mod.has_check? &&
-      is_session_type_compat?(mod) &&
-      is_module_platform?(mod) &&
-      is_module_arch?(mod) &&
+    return false unless mod.kind_of?(Msf::Exploit::Local) && mod.has_check?
+
+    is_session_type_compat?(mod) &&
+      (@validate_platform ? is_module_platform?(mod) : true) &&
+      (@validate_arch ? is_module_arch?(mod) : true) &&
       is_module_options_ready?(mod)
-    )
   end
 
   def setup
     return unless session
 
     print_status "Collecting local exploits for #{session.session_type}..."
 
+    setup_validation_options
+    setup_color_options
+
     # Collects exploits into an array
     @local_exploits = []
-    framework.exploits.each do |name, _obj|
+    framework.exploits.each_with_index do |(name, _obj), index|
+      @valid_reasons = []
+
+      print "%bld%blu[*]%clr Gathering exploit #{index + 1} / #{framework.exploits.count}\r"
```

Comment:
```suggestion
      print "%bld%blu[*]%clr Collecting exploit #{index + 1} / #{framework.exploits.count}\r"
```

Probably best to use the same verbage as the before the loop to not confuse users